### PR TITLE
fix: binary installation, checksums, cleanup

### PR DESCRIPTION
**Pull request by plyr4**
⚠️ required changes for terraform v >= 1.8, see: https://github.com/hashicorp/terraform/issues/35075

- adds checksum validation to the Dockerfile's default terraform binary installation
- ~removes `PHONY` references in make~ (redacted)
- migrates from go-getter to [hc-install](https://github.com/hashicorp/hc-install) based on recommendations from hashicorp over breaking changes made to the way terraform is zipped and released.

inspiration for code changes are from https://github.com/runatlantis/atlantis/pull/4494

### CODE REVIEW

**Review by timhuynh94 — DISMISSED** [review]

**Review by wass3rw3rk — COMMENTED** [review]
why the `PHONY` removal? i understand the chance we have a file named any of the targets is low (in which case i believe it would do nothing), but it's used to indicate non-file targets and is consistent with other repos. it adds minimal noise and is technically correct?

one could open the can-o-worms of should we even use Makefiles to start with?

**Comment by plyr4** [comment]
> why the `PHONY` removal? i understand the chance we have a file named any of the targets is low (in which case i believe it would do nothing), but it's used to indicate non-file targets and is consistent with other repos. it adds minimal noise and is technically correct?
> 
> one could open the can-o-worms of should we even use Makefiles to start with?

it was an item on our internal roadmap to consolidate and remove phony usage to bring these all to a consistent standard. 
i dont know who recommended it initially.

dont shoot the messenger!

**Comment by plyr4** [comment]
i removed the removal of phony. to be honest i dont care if its there, i was just following blindly.
i'd rather not muddle this PR which is actually fixing something. thanks for the review, i appreciate it

File: Dockerfile [hunk 1]
```diff
@@ -13,9 +13,18 @@ FROM alpine:latest@sha256:0a4eaa0eecf5f8c050e5bba433f58c052be7587ee8af3e8b3910ef
...
+ENV CHECKSUM_FILE="SHA256SUMS"
+
+# download and verify the Terraform binary
+RUN wget -q "${TERRAFORM_ZIP}" -O terraform.zip && \
+    wget -q "${CHECKSUM_URL}" -O "${CHECKSUM_FILE}" && \
+    EXPECTED_CHECKSUM=$(grep "terraform_${TERRAFORM_VERSION}_linux_amd64.zip" "${CHECKSUM_FILE}" | awk '{ print $1 }') && \
+    ACTUAL_CHECKSUM=$(sha256sum terraform.zip | awk '{ print $1 }') && \
+    if [ "$EXPECTED_CHECKSUM" != "$ACTUAL_CHECKSUM" ]; then echo "Checksum verification failed"; exit 1; fi && \
+    unzip terraform.zip -d /bin && \
+    rm -f terraform.zip "${CHECKSUM_FILE}"
```
Comment (wass3rw3rk) [inline Dockerfile:27]: for consideration: if we left the filename intact, after getting the checksum file, we could just use `sha256sum --check --ignore-missing "${CHECKSUM_FILE}"` and check the exit code. less shell scripting.

**Review by wass3rw3rk — APPROVED** [review]
love the checksum check addition. one suggestion. since it works as-is, i'll approve. happy to re-approve if you decide to take it up.

File: Dockerfile (hunk 1 above)
Comment (plyr4) [reply Dockerfile]: good call! see https://github.com/go-vela/vela-terraform/pull/181/commits/099810069fd56346101c2b49f21435edd42a11cc

**Review by wass3rw3rk — APPROVED** [review]

**Pull request merged**